extensions/amp-analytics/0.1/activity-impl.js

```diff
@@ -106,6 +116,21 @@ class ActivityHistory {
     }
     return totalEngagedTime;
   }
+
+  /**
+   * Get the incremental engaged time since the last push and reset it.
+   * @param {string} name
+   * @return {number}
+   */
+  getIncrementalEngagedTime(name = '') {
+    if (!this.incrementalEngagedTime_.hasOwnProperty[name]) {
+      this.incrementalEngagedTime_[name] = this.totalEngagedTime_;
```

Comment:
Will need to take `this.prevActivityEvent_` into consideration as well. 
In that case, how do you think of moving the logic to class `Activity` instead?

Reply:
Since I'm piggybacking on `this.totalEngagedTime_` which already does take `this.prevActivityEvent_` into consideration, I don't think I need to. Am I misunderstanding?

My intention in this function was to check the existence of `this.incrementalEngagedTime_[name]` and if it doesn't exist then initialize it with the `this.totalEngagedTime_` value and return that. If it does exist put it's value in a temporary variable, this value should be the incremental engaged time that was sent in the previous heartbeat. Reset `this.incrementalEngagedTime_[name]` to the current `this.totalEngagedTime_` and return the difference between the `currentIncrementalEnagagedTime` (the previous heartbeat engaged time) and the `this.totalEngagedTime_`. Which should be the engaged time that has passed since the last heartbeat. Right?

Reply:
It's true that `this.totalEngagedTime_` would take `this.prevActivityEvent_` into consideration [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/activity-impl.js#L89). 
But that's only with `ActivityHistory.push`. When we try to call `#getTotalEngagedTime` [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/activity-impl.js#L101), we still need to take the `this.prevActivityEvent_` into consideration again it can be in the middle of an active event. Think of a special case, that an activity event never ends. 
Thus, instead of using the `this.totalEngagedTime_`, I think storing the return value of last `#getTotalEngagedTime()` call is more accurate. That's why I suggest moving the logic to `Activity` class.

Make sense?